java/connector-node/risingwave-sink-es/src/main/java/com/risingwave/connector/EsSink.java

```diff
@@ -0,0 +1,264 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.risingwave.connector;
+
+import com.risingwave.connector.api.TableSchema;
+import com.risingwave.connector.api.sink.SinkBase;
+import com.risingwave.connector.api.sink.SinkRow;
+import io.grpc.Status;
+import java.io.IOException;
+import java.util.*;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.bulk.BackoffPolicy;
+import org.elasticsearch.action.bulk.BulkProcessor;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.TimeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * Note:
+ *
+ * 1. TODO: If no primary key is defined on the DDL, the connector can only operate in append
+ * mode for exchanging INSERT only messages with external system.
+ *
+ * 2. Currently, index is fixed.
+ *
+ * 3. Possible settings from flink:
+ * https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/connectors/table/elasticsearch/
+ *
+ * 4. bulkprocessor and high-level-client are deprecated in es 8 java api.
+ */
+public class EsSink extends SinkBase {
+    private static final String ERROR_REPORT_TEMPLATE = "Error when exec %s, message %s";
+
+    private final EsSinkConfig config;
+    private final BulkProcessor bulkProcessor;
+    private final RestHighLevelClient client;
+    private static final Logger LOG = LoggerFactory.getLogger(EsSink.class);
```

Comment:
nit: usually the `LOG` or static final field is put at the beginning of the class.